retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactory.java

```diff
@@ -157,6 +173,10 @@ private RxJavaCallAdapterFactory(Scheduler scheduler) {
         subscriber.onCompleted();
       }
     }
+
+    @Override public void call() {
+      call.cancel();
```

Comment:
Yeah. I'm also reusing the object for multiple things to save allocations
and instance size which adds to the responsibility and naming confusion.

On Tue, Mar 29, 2016, 9:42 PM Jesse Wilson <notifications@github.com> wrote:

> In
> retrofit-adapters/rxjava/src/main/java/retrofit2/adapter/rxjava/RxJavaCallAdapterFactory.java
> <https://github.com/square/retrofit/pull/1702#discussion_r57827300>:
>
> > @@ -157,6 +173,10 @@ private RxJavaCallAdapterFactory(Scheduler scheduler) {
> >          subscriber.onCompleted();
> >        }
> >      }
> > +
> > +    @Override public void call() {
> > +      call.cancel();
>
> Not your code, but Rx makes this really counterintuitive
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/square/retrofit/pull/1702/files/c107778914815466bda00878c4cbc6b799a6e133#r57827300>
>